src/mds/CInode.cc

```diff
@@ -703,41 +703,35 @@ std::pair<bool, std::vector<CDir*>> CInode::get_dirfrags_under(frag_t fg)
   auto& all = result.first;
   auto& dirs = result.second;
 
-  {
-    frag_vec_t leaves;
-    dirfragtree.get_leaves_under(fg, leaves);
-    for (const auto &leaf : leaves) {
-      if (auto it = dirfrags.find(leaf); it != dirfrags.end()) {
-        dirs.push_back(it->second);
-      } else {
-        all = false;
-      }
-    }
-  }
-
-  if (all)
+  all = false;
+  frag_vec_t children;
+  if (dirfrags.count(fg)){
```

Comment:
if (auto it = dirfrags.find(fg); it != dirfrags.end()) {
   dirs.push_back(it->second());
   return result;
}